### PR TITLE
Fix double-scroll issue on iOS devices

### LINKED ISSUE #2683: Hidden header as scrolling down
## Description

Once landing in the search results, I can scroll down in the header area to hide it and reveal a white space below the footer. The bug makes openverse look like inserted as an iframe.

**Edit:** Updated in the comments below:

> I no longer see the bug in MacOS. In iOS, is now only visible on Safari.

## Reproduction

1. Go to openverse.org 
2. Search something and land on the search results
3. Scroll down within the header area
4. See the error.

### Safari on iOS
https://github.com/WordPress/openverse/assets/895819/18ce9071-a13a-4ab5-82fd-7d9545a4450f

### Safar on MacOS
https://github.com/WordPress/openverse/assets/895819/58fc5e34-92ad-4ba8-90b3-5e48bdbf6cdb

## Environment

- Device: iPhone 12 pro and Macbook pro M1 14-inch 
- OS: iOS 16.5.1 and Macos Ventura 13.4.1
- Browser:
   - in iOS: Safari and Firefox
   - in Macbook: Safari and Chrome in all breakpoints. In Firefox the bug is not visible 

## Additional context

I think it relates to #2596

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Fixes #2683 by @fcoveram

## Description

This PR fixes the janky scrolling behaviour on iOS by fixing the order in which the `100vh` and `100dvh` heights are applied. On iOS they were being applied in the opposite order of what's desired, leading to the problem.

Fixing the order as `100vh`, `100dvh` ensures that `dvh` wins, if supported, fixing the issue.

## Testing Instructions

0. Install XCode on your Mac (it's huge, like ~8GB so do not use cellular or limited plans)
0. Check out the PR and run the dev server.
0. Visit https://localhost:8443 from Safari in the simulator.
1. Try (and hopefully fail) to reproduce the problem described in #2683.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by dhruvkb** [comment]
Reviewers changed to folks who own iPhones or MacBooks.

File: frontend/tailwind.config.js [hunk 1]
```diff
@@ -197,6 +197,12 @@ module.exports = {
       serif: [...defaultTheme.fontFamily.serif],
     },
     extend: {
+      height: {
```
Comment (obulat) [inline frontend/tailwind.config.js:200]: Nice abstraction!

**Review by obulat — APPROVED** [review]
I tested using an iPhone emulator, and the bug when you couldn't see the top of the header  when you navigate from the homepage to the search page is fixed - so approving to unblock this.

I think we also need to add `dyn-screen` to the `default` layout. Also, in the emulator, I can see a white space in the bottom of the homepage. Can you see it on an iphone, @dhruvkb ?

Screenshot 2023-07-27 at 11 28 53 AM

**Comment by dhruvkb** [comment]
@obulat that white space is likely because of the absence of the soft keyboard. It's not present on the actual phone.

**Comment by obulat** [comment]
> @obulat that white space is likely because of the absence of the soft keyboard. It's not present on the actual phone.

Nice! So happy that it's not a real problem :)

**Review by zackkrida — APPROVED** [review]
@obulat is right that we need to add this to the homepage, either here or another PR is fine. The bug in question is fixed. Tested in the simulator and on an actual iPhone 12.

**Comment by dhruvkb** [comment]
I've updated the PR to include the changes to the `default` and `content-layout` layouts. Will merge when tests pass.

**Review by fcoveram — APPROVED** [review]
It works great now ✨ 🚀

**Comment by dhruvkb** [comment]
Update regarding the white space that appears below the page on input focus: That seems to be a Safari thing that cannot be disabled or worked around. Confirmed this to also affect Wikipedia and DuckDuckGo. Other iOS browsers like Arc or Quiche do not suffer from this bug.

**Pull request merged**